### ISSUE TITLE
Debugging context menu completely lock-up my OS UI

### ISSUE THREAD

**Issue by laeubi** [eclipse-platform/eclipse.platform.debug#26]
Version: 2022-03 (4.23.0)
Build id: 20220310-1457

Debian 4.19.194-3 (2021-07-18) x86_64 GNU/Linux

### Steps to reproduce: 

# SAVE ALL YOUR WORK AS YOU PROBABLY NEED TO KILL YOUR WINDOW MANAGER OR RESTART SYSTEM

1. Go to the debug perspective
2. add a break point inside the constructor of `org.eclipse.debug.internal.ui.actions.LaunchShortcutAction.LaunchShortcutAction(String, LaunchShortcutExtension)`
3. Start a new debugged Eclipse Application
4. create a new java project with a main class
5. righclick and navigate to the "Run As" menu

Now for me the following happens:

- Breakpoint is triggered and the Host-Eclipse is bring to foreground
- I see the popup menu of the debugged eclipse
- my system is locked-up, I can not use any application
- I need to kill the window manager to get out of this state killing all UI applications

**Comment by iloveeclipse** [comment]
This is GTK issue. Debugging popup menus freezes GTK because it tries to activate pupup and debugger window at same time. Was always there and if you want fix it, you probably should report a bug for GTK.

**Comment by laeubi** [comment]
Any chance the debugger can detect this and simply throw "debugging menus is bad" or something?

**Comment by iloveeclipse** [comment]
One would need to traverse thread stack on every breakpoint hit and try to detect if we are in the SWC specific menu part.
I believe it is doable but would add overhead for every breakpoint hit, not sure if that is something to be worried or not.
Would also need to restrict that to "Eclipse" launch config debugging only.

**Comment by Bananeweizen** [comment]
Side note for Windows users: A similar problem exists there. Debugging menu code can lead to not being able to switch away from the debugged application, at least if it consumes the whole desk. The simple workaround is to have IDE and debugged application side by side, since the _visible_ UI of the host IDE remains usable (e.g. to kill the debugged app).

**Comment by iloveeclipse** [comment]
Yeah, on Windows it "seems" to lock up, but after clicking on the debugger/debuggee taskbar entries few times one can get the desktop running again. 

On Linux KDE the only (local) workaround is to switch to the text console (ALT+7 I believe, haven't used long time since working in vnc), & kill the debuggee process, or ssh / vnc to the locked machine from another one and here also kill the debuggee process. One gets used to it :-)
